src/nnfusion/engine/pass/graph/autodiff/roll.cpp

```diff
@@ -0,0 +1,33 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+#include "backward_registry.hpp"
+
+REGISTER_BACKWARD_TRANSLATOR(Roll).translator(
+    [](std::shared_ptr<GNode> forward_node,
+       const GNodeIndexVector& outputs_grad,
+       std::shared_ptr<nnfusion::graph::Graph> graph) -> GNodeIndexVector {
+        NNFUSION_CHECK(outputs_grad.size() == 1) << "Roll have only 1 output, but "
+                                                 << outputs_grad.size() << " outputs_grad provided";
+        // y = roll(x, shifts, dims)
+        // x_grad = roll(f_grad, fmap(reverse_list(shifts), [](int64_t i){return -i;}), reverse_list(dims))
```

Comment:
🟢 [Option]

Remove unused code instead of commenting them

Reply:
Done.